openapi_metadata_test.go

```diff
@@ -0,0 +1,314 @@
+	var parsers MetadataParsers
+
+	require.NotNil(t, &parsers)
+	require.NotNil(t, parsers.registeredParsers)
+	require.Empty(t, parsers.registeredParsers)
+	require.NotNil(t, parsers.registeredNames)
+	require.Empty(t, parsers.registeredNames)
+}
+
+// Return all registered metadata parsers from the internal slice
+func TestGetRegisteredParsersReturnsAllParsers(t *testing.T) {
+	mp := &MetadataParsers{
+		registeredParsers: []MetadataParserEntry{
+			{Name: "parser1", Parser: func(params MetadataParserParams) { return }},
+			{Name: "parser2", Parser: func(params MetadataParserParams) { return }},
+		},
+	}
+
+	parsers := mp.GetRegisteredParsers()
+
+	require.NotNil(t, parsers)
+	require.Len(t, parsers, 2)
+
+	assert.Equal(t, "parser1", parsers[0].Name)
+	assert.Equal(t, "parser2", parsers[1].Name)
+}
+
+// Return empty slice when no parsers are registered
+func TestGetRegisteredParsersReturnsEmptySlice(t *testing.T) {
+	mp := &MetadataParsers{
+		registeredParsers: []MetadataParserEntry{},
+	}
+
+	parsers := mp.GetRegisteredParsers()
+
+	require.NotNil(t, parsers)
+	require.Empty(t, parsers)
+}
+
+// Verify concurrent access is properly synchronized using mutex lock
+func TestGetRegisteredParsersConcurrentAccess(t *testing.T) {
+	mp := &MetadataParsers{
+		registeredParsers: []MetadataParserEntry{
+			{Name: "parser1", Parser: func(params MetadataParserParams) { return }},
+			{Name: "parser2", Parser: func(params MetadataParserParams) { return }},
+		},
+	}
+
+	var wg sync.WaitGroup
+	numGoroutines := 3
+	wg.Add(numGoroutines)
+
+	for i := 0; i < numGoroutines; i++ {
+		go func() {
+			defer wg.Done()
+			parsers := mp.GetRegisteredParsers()
+
+			require.NotNil(t, parsers)
+			require.Len(t, parsers, 2)
+		}()
+	}
+
+	wg.Wait()
+}
+
+// Check that returned slice contains expected parser entries
+func TestGetRegisteredParsersContainsExpectedEntries(t *testing.T) {
+	mp := &MetadataParsers{
+		registeredParsers: []MetadataParserEntry{
+			{Name: "expectedParser1", Parser: func(params MetadataParserParams) { return }},
+			{Name: "expectedParser2", Parser: func(params MetadataParserParams) { return }},
+		},
+	}
+
+	parsers := mp.GetRegisteredParsers()
+
+	require.NotNil(t, parsers)
+	require.Len(t, parsers, 2)
+
+	require.Equal(t, "expectedParser1", parsers[0].Name)
+	require.Equal(t, "expectedParser2", parsers[1].Name)
+}
+
+// Verify returned slice matches original registered parsers
+func TestGetRegisteredParsersMatchesOriginal(t *testing.T) {
+	mp := &MetadataParsers{
+		registeredParsers: []MetadataParserEntry{
+			{Name: "parser1", Parser: func(params MetadataParserParams) { return }},
+			{Name: "parser2", Parser: func(params MetadataParserParams) { return }},
+		},
+	}
+
+	parsers := mp.GetRegisteredParsers()
+
+	require.NotNil(t, parsers)
+	require.Len(t, parsers, 2)
+
+	if !reflect.DeepEqual(parsers, mp.registeredParsers) {
+		t.Errorf("Returned parsers do not match the original registered parsers")
+	}
+}
```

Comment:
Since `mp.GetRegisteredParsers()` is returning the `mp.registeredParsers`, this test is useless. What do you wanted to test exactly?